operators/tokenizer/bpe_kernels.cc

```diff
@@ -552,13 +554,140 @@ SpmTokenizer::SpmTokenizer()
 
 JsonFastTokenizer::JsonFastTokenizer() : KernelBpeTokenizer(kGPT2Configuration) {}
 
+/*
+Read more here: https://github.com/huggingface/transformers/blob/60bb571e993b7d73257fb64044726b569fef9403/src/transformers/convert_slow_tokenizer.py#L1454
+
+Note: this is similar to the BPE CreateByteEncoder, however for decoding the .tiktoken bytes
+we need to store the strings rather than their IDs, and thereby need a separate map.
+*/
+void JsonFastTokenizer::CreateUnicodeByteEncoder() {
```

Comment:
what's the difference from BpeModel::CreateByteEncoder?

Reply:
BpeModel::CreateByteEncoder creates a map where the values are the token ids (uint32s) and uses the input ids for this map, e.g. `byte_encoder_[i] = GetTokenId(ustring::EncodeUTF8Char(i))`.

CreateUnicodeByteEncoder creates a map where the values are the token strings, and needs to be used just to convert bytes to string (including unicodes), and therefore does not need to get the token ids (which it also cannot at that step because we need to process the tiktoken file first).